lib/brakeman/checks/check_execute.rb

```diff
@@ -143,7 +143,11 @@ def dangerous? exp
       next if SAFE_VALUES.include? e
       next if shell_escape? e
 
-      if node_type? e, :or, :evstr, :dstr
+      if node_type? e, :if
+        if res = dangerous?(e.values[1..-1])
```

Comment:
I would prefer

`dangerous?(e.then_clause) or dangerous(e.else_clause)`

Reply:
Ah neat, didn't know about those methods. Will change!

Reply:
Actually, this doesn't seem to work, I believe because `dangerous?` calls `each_sexp` on its argument and those clauses aren't "high-level" enough in the tree to have that work as we want. Feel free to fiddle around with the code in case there's a way to do it that I missed though!